internal/cli/cmd/deployplan.go

```diff
@@ -55,6 +57,10 @@ func NewDeployPlanCmd() *cobra.Command {
 			return err
 		}
 
+		if getDeployReason(opts) == "" && deploy.RequireReason(config) {
```

Comment:
Same here, deploy.RequireReason() first, as that's what's triggering the condition.